pkg/controller/cluster/cluster_controller.go

```diff
@@ -498,6 +499,8 @@ func (c *Controller) manageMachineSets(machineSets []*clusteroperator.MachineSet
 		}
 		if deleteMachineSet {
 			machineSetsToDelete = append(machineSetsToDelete, clusterMachineSets[i])
+		} else if updateMachineSet {
+			machineSetsToUpdate = append(machineSetsToUpdate, clusterMachineSets[i])
```

Comment:
Replace `clusterMachineSets[i]` with `machineSetToCreate`. The former is what is currently in storage; the latter is what we want to update to. It would probably behoove us to rename `machineSetToCreate`, too, since it is not used only for creating a machine set.

Reply:
Yes, I broke this with a change after I had tested it. Will fix and add the relevant unit test.